core/routemgmt/activateApi.js

```diff
@@ -0,0 +1,383 @@
+  }
+  if (docid) params.docid = docid;
+  console.log('getApiDoc() for: ', params);
+    return whisk.invoke({
+      name: actionName,
+      blocking: true,
+      parameters: params
+    })
+    .then(function (activation) {
+      console.log('whisk.invoke('+actionName+', '+params.namespace+', '+params.basepath+') ok');
+      console.log('Results: '+JSON.stringify(activation));
+    if (activation && activation.result && activation.result.apis &&
+        activation.result.apis.length > 0 && activation.result.apis[0].value &&
+        activation.result.apis[0].value._rev) {
+        return Promise.resolve(activation.result.apis[0].value);
+      } else {
+        console.error('Invalid API doc returned!');
+        return Promise.reject('Document for namepace \"'+namespace+'\" and basepath \"'+basepath+'\" was not located');
+      }
+    })
+    .catch(function (error) {
+      console.error('whisk.invoke('+actionName+', '+params.namespace+', '+params.basepath+') error: '+JSON.stringify(error));
+      return Promise.reject(error);
+    });
+}
+
+function addTenantToGateway(gwInfo, namespace) {
+  var options = {
+    url: gwInfo.gwUrl+'/tenants',
+    agentOptions: {rejectUnauthorized: false},
+    headers: {
+      'Content-Type': 'application/json',
+      'Accept': 'application/json'
+    },
+    json: {
+      instance: 'openwhisk',    // Use a fixed instance so all openwhisk tenants have a common instance
+      namespace: namespace
+    }
+  };
+  if (gwInfo.gwAuth) {
+    options.headers.Authorization = 'Basic ' + gwInfo.gwAuth;
+  }
+  console.log('addTenantToGateway: request: '+JSON.stringify(options));
+
+  return new Promise(function(resolve, reject) {
+    request.put(options, function(error, response, body) {
+      var statusCode = response ? response.statusCode : undefined;
+      console.log('addTenantToGateway: response status: '+ statusCode);
+      error && console.error('Warning: addTenantToGateway request failed: '+JSON.stringify(error));
+      body && console.log('addTenantToGateway: response body: '+JSON.stringify(body));
+
+      if (error) {
+        console.error('addTenantToGateway: Unable to configure a tennant on the API Gateway: '+JSON.stringify(error))
+        reject('Unable to configure the API Gateway: '+JSON.stringify(error));
+      } else if (statusCode != 200) {
+        console.error('addTenantToGateway: failure: response code: '+statusCode)
+        reject('Unable to configure the API Gateway: Response failure code: '+statusCode);
+      } else {
+        if (body && body.id) {  // body has format like:  { id: GUID, namespace: NAMESPACE, instance: 'openwhisk' }
+          console.log('addTenantToGateway: got a single tenant resposne');
+          resolve(body);
+        } else {
+          console.error('addTenantToGateway: failure: No tenant guid provided')
+          reject('Unable to configure the API Gateway: Invalid response from API Gateway');
+        }
+      }
+    });
+  });
```

Comment:
I agree.  Use of the request-promise package would add a bit more readability.  Alas, that issue is not currently resolved; but I'm thinking that it should not block the initial API GW code delivery.